src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let flag = 0;
+  let sum = 0;
+
+  const makeSum = (a) => {
+    if (typeof a === 'number') {
+      sum += a;
+      flag = 0;
+
+      return makeSum;
+    } else if (typeof a !== 'number') {
+      flag++;
+
+      sum = flag === 2 ? 0 : sum;
+
+      return sum;
```

Comment:
Returning 'sum' here is incorrect because it will return the sum whenever a non-number is passed, which is not the intended behavior. The function should return the sum only when called without any arguments.